provider/pkg/cf2pulumi/renderer.go

```diff
@@ -898,12 +898,17 @@ func (ctx *renderContext) renderResource(attr *ast.MappingValueNode) (model.Body
 				return nil, diagnostics, fmt.Errorf("'%v' must be a mapping", keyString(f))
 			}
 			for _, sf := range subValues {
+				keyValue := keyString(sf)
+				// Properties should never contain a colon. This is only used within metadata which we discard.
```

Comment:
I'm a little nervous that we'll find out this assumption is not true in the future, but I'm OK with pressing forward since we know it fixes one real world use case and don't know of any it would break.